fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckNetworkMiddleware.kt

```diff
@@ -125,10 +129,14 @@ class ReviewQualityCheckNetworkMiddleware(
         }
     }
 
-    private suspend fun pollForAnalysisStatus(): AnalysisStatusDto? =
+    private suspend fun pollForAnalysisStatus(onEachSuccessfulPoll: (progress: Double) -> Unit): AnalysisStatusProgressDto? =
         retry(
-            predicate = { it.isPendingOrInProgress() },
-            block = { reviewQualityCheckService.analysisStatus() },
+            predicate = { it?.status.isPendingOrInProgress() },
+            block = {
+                reviewQualityCheckService.analysisStatus()?.also {
+                    onEachSuccessfulPoll(it.progress!!)
```

Comment:
We don't need the double bangs here, we're making sure this function is only called when `analysisStatus()` is returning not null and `progress` is non nullable.

Reply:
Gotcha, Just to clarify, progress is non nullable, but on the UI side, progress is null by default.

Reply:
Yeah, that's right 👍